Basic-VoIP-Call-Java/app/src/main/java/com/tokbox/sample/basicvoipcall/MainActivity.java

```diff
@@ -0,0 +1,251 @@
+package com.tokbox.sample.basicvoipcall;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.telecom.PhoneAccount;
+import android.telecom.PhoneAccountHandle;
+import android.telecom.TelecomManager;
+import android.telecom.VideoProfile;
+import android.telephony.TelephonyManager;
+import android.util.Log;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+
+import com.opentok.android.AudioDeviceManager;
+
+import java.util.List;
+
+import pub.devrel.easypermissions.AfterPermissionGranted;
+import pub.devrel.easypermissions.EasyPermissions;
+
+//import com.google.firebase.messaging.FirebaseMessaging;
+
+public class MainActivity extends AppCompatActivity implements EasyPermissions.PermissionCallbacks {
+
+    private static final String TAG = MainActivity.class.getSimpleName();
+
+    private static final int PERMISSIONS_REQUEST_CODE = 124;
+
+    private RelativeLayout publisherViewContainer;
+    private LinearLayout subscriberViewContainer;
+
+
+    private TelecomManager mTelecomManager;
+    private TelephonyManager mTelephonyManager;
+    private PhoneAccountHandle mPhoneAccountHandle;
+    private PhoneAccount mPhoneAccount;
+    //private OTFireBaseMessagingService mOTFireBaseMessagingService;
+
+    @SuppressLint("NewApi")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        if (!OpenTokConfig.isValid()) {
+            finishWithMessage("Invalid OpenTokConfig. " + OpenTokConfig.getDescription());
+            return;
+        }
+
+        requestPermissions();
+
+        /*
```

Comment:
If this commented out code should stay, maybe we should have a comment explaining why.

Reply:
Please see my other comment.